httpclient/handle.go

```diff
@@ -23,11 +23,25 @@ func handle(w http.ResponseWriter, r *http.Request) {
 
 	cfg, err := configparse.JSON(b)
 	if err != nil {
-		internalError(w, err)
+		clientError(w, err)
 		return
 	}
 
 	rep, err := runner.New(streamProgress(w)).Run(r.Context(), cfg)
+	var invalidConfigError *runner.InvalidConfigError
+	switch {
+	case err == runner.ErrCanceled:
+		clientError(w, err)
+		return
+	case errors.As(err, &invalidConfigError):
+		clientError(w, invalidConfigError.Errors...)
+	case err == nil:
+		// Pass through.
```

Comment:
Nice alternative to nesting the `switch` block inside an `if err != nil` block.
Can we just move this case to the top so it's not in the middle of other cases where we actually handle errors?